vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,510 @@
+```{r}
+rep_tri <- rep_tri_df |>
+  select(-reference_date) |>
+  as.matrix()
+```
+## Specify the `baselinenowcast` model
+
+We will use the helper function `allocate_reference_times()` to determine how many reference times to use for delay estimation and uncertainty estimation.
+We will use the defaults for this function, which uses 3 times the maximum delay as the total amount of training volume, split evenly between reference times used for delay estimation and reference times used for producing retrospective nowcasts for uncertainty estimation.
+In this case, this will result in 37 reference times used for delay estimation, and 38 used for uncertainty estimation.
+```{r}
+ref_time_allocation <- allocate_reference_times(rep_tri)
+n_history_delay <- ref_time_allocation$n_history_delay
+n_retrospective_nowcasts <- ref_time_allocation$n_retrospective_nowcasts
+```
+
+# Run the `baselinenowcast` workflow
+Here we demonstrate the workflow using the intermediate-level functions. For more details, see the [Getting Started](baselinenowcast.html) vignette.
+
+First, we generate the point estimate of the nowcast as a "filled in" reporting triangle
+```{r}
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
+```
+
+Next, we estimate and apply uncertainty by generating retrospective nowcasts and comparing against later observed data.
+```{r}
+nowcast_draws_df <- estimate_and_apply_uncertainty(
+  pt_nowcast_matrix,
+  rep_tri,
+  n_history_delay = n_history_delay,
+  n_retrospective_nowcasts = n_retrospective_nowcasts,
+  draws = 100
+)
+
+head(nowcast_draws_df)
+```
+
+# Summarise and plot the nowcast
+Now that we have a dataframe of probabilistic nowcast draws for all reference times in our original reporting triangle, we can summarise the median and 50th and 95th quantiles.
+```{r}
+nowcast_summary_df <-
+  nowcast_draws_df |>
+  group_by(time) |>
+  summarise(
+    median = median(pred_count),
+    q50th_lb = quantile(pred_count, 0.25),
+    q50th_ub = quantile(pred_count, 0.75),
+    q95th_lb = quantile(pred_count, 0.025),
+    q95th_ub = quantile(pred_count, 0.975)
+  )
+```
+
+Next, we will join the initially reported data and the final evaluation data, both summarised by reference time.
+```{r}
+training_df_by_ref_date <- training_df |>
+  filter(report_date <= nowcast_date) |>
+  group_by(reference_date) |>
+  summarise(initial_count = sum(count)) |>
+  mutate(time = row_number())
+
+eval_data <- long_df |>
+  filter(
+    delay <= max_delay,
+    reference_date <= nowcast_date
+  ) |>
+  group_by(reference_date) |>
+  summarise(final_count = sum(count)) |>
+  mutate(time = row_number()) |>
+  select(time, final_count)
+
+nowcast_w_data <- nowcast_summary_df |>
+  left_join(training_df_by_ref_date,
+    by = "time"
+  ) |>
+  left_join(eval_data,
+    by = "time"
+  )
+head(nowcast_w_data)
+```
+
+# Plot against later observed "final" data
+Next we will make a plot of the nowcast
```

Comment:
I dont think we hav etold anyone what a nowcast is or why they need one. Or visualised the data with truncation until now?

Reply:
have added at beginning and throughout this section